server/blob/blob-cassandra/src/main/java/org/apache/james/blob/cassandra/CassandraBlobStore.java

```diff
@@ -19,203 +19,127 @@
 
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
 import java.io.InputStream;
-import java.nio.ByteBuffer;
-import java.util.Comparator;
-import java.util.List;
-import java.util.NoSuchElementException;
-import java.util.stream.Collectors;
 
 import javax.inject.Inject;
 
 import org.apache.commons.io.IOUtils;
-import org.apache.commons.lang3.tuple.Pair;
 import org.apache.james.backends.cassandra.init.configuration.CassandraConfiguration;
 import org.apache.james.blob.api.BlobId;
 import org.apache.james.blob.api.BlobStore;
 import org.apache.james.blob.api.BucketName;
 import org.apache.james.blob.api.HashBlobId;
-import org.apache.james.blob.api.ObjectNotFoundException;
-import org.apache.james.blob.cassandra.utils.DataChunker;
-import org.apache.james.util.ReactorUtils;
 
 import com.datastax.driver.core.Session;
 import com.google.common.annotations.VisibleForTesting;
 import com.google.common.base.Preconditions;
-import reactor.core.publisher.Flux;
+import com.google.common.hash.Hashing;
+import com.google.common.hash.HashingInputStream;
+import com.google.common.io.FileBackedOutputStream;
 import reactor.core.publisher.Mono;
-import reactor.core.scheduler.Schedulers;
-import reactor.util.function.Tuple2;
+import reactor.util.function.Tuples;
 
 public class CassandraBlobStore implements BlobStore {
 
-    private static final int PREFETCH = 16;
-    private static final int MAX_CONCURRENCY = 1;
-    private final CassandraDefaultBucketDAO defaultBucketDAO;
-    private final CassandraBucketDAO bucketDAO;
-    private final DataChunker dataChunker;
-    private final CassandraConfiguration configuration;
+    public static final boolean LAZY_RESSOURCE_CLEANUP = false;
+    public static final int FILE_THRESHOLD = 10000;
     private final HashBlobId.Factory blobIdFactory;
+    private final BucketName defaultBucketName;
+    private final CassandraDumbBlobStore dumbBlobStore;
 
     @Inject
-    CassandraBlobStore(CassandraDefaultBucketDAO defaultBucketDAO, CassandraBucketDAO bucketDAO, CassandraConfiguration cassandraConfiguration, HashBlobId.Factory blobIdFactory) {
-        this.defaultBucketDAO = defaultBucketDAO;
-        this.bucketDAO = bucketDAO;
-        this.configuration = cassandraConfiguration;
+    CassandraBlobStore(HashBlobId.Factory blobIdFactory, BucketName defaultBucketName, CassandraDumbBlobStore dumbBlobStore) {
         this.blobIdFactory = blobIdFactory;
-        this.dataChunker = new DataChunker();
+        this.defaultBucketName = defaultBucketName;
+        this.dumbBlobStore = dumbBlobStore;
     }
 
     @VisibleForTesting
-    public CassandraBlobStore(Session session) {
-        this(new CassandraDefaultBucketDAO(session),
-            new CassandraBucketDAO(new HashBlobId.Factory(), session),
-            CassandraConfiguration.DEFAULT_CONFIGURATION,
-            new HashBlobId.Factory());
+    public static CassandraBlobStore forTesting(Session session) {
```

Comment:
Is it ok to declare testing static methods in a main class like that?

Reply:
I personnaly concider it OK as the intent in the naming is explicit.